x-pack/plugin/enrich/src/test/java/org/elasticsearch/xpack/enrich/EnrichProcessorFactoryTests.java

```diff
@@ -169,20 +170,14 @@ public void testNonExistingDecorateField() throws Exception {
         assertThat(e.getMessage(), equalTo("source field [rank] does not exist in policy [majestic]"));
     }
 
-    private static Supplier<ClusterState> createClusterStateSupplier(String policyName, EnrichPolicy policy) {
-        EnrichMetadata enrichMetadata = new EnrichMetadata(Map.of(policyName, policy));
-        ClusterState state = ClusterState.builder(new ClusterName("_name"))
-            .metaData(MetaData.builder().putCustom(EnrichMetadata.TYPE, enrichMetadata).build())
-            .build();
-        return () -> state;
-    }
-
-    private static Supplier<ClusterState> createClusterStateSupplier() {
-        EnrichMetadata enrichMetadata = new EnrichMetadata(Map.of());
-        ClusterState state = ClusterState.builder(new ClusterName("_name"))
-            .metaData(MetaData.builder().putCustom(EnrichMetadata.TYPE, enrichMetadata).build())
-            .build();
-        return () -> state;
+    private static Function<String, EnrichPolicy> policyLookup(String policyName, EnrichPolicy policy) {
+        return name -> {
+            if (name.equals(policyName)) {
+                return policy;
+            } else {
+                return null;
+            }
+        };
```

Comment:
Can you add tests that changes the policies but don't change the pipeline's config ?

Reply:
+1